File: Java Script - Introduction to Programming/8.1. Exam Preparation - Part I/09. Perfect Diamond.js
function perfectDiamond(n) {
  n = +n

  if (n === 1)
    return ' '.repeat(n - 1) + '*' + ' '.repeat(n - 1)

  //top
  console.log(' '.repeat(n - 1) + '*' + ' '.repeat(n - 1));

  //before middle
  for (let i = 1; i <= n - 2; i++) {
    console.log(' '.repeat(n - i - 1) + ('*' + '-').repeat(i) + '*' + ' '.repeat(n - i - 1));
  }

  //middle
  console.log(('*' + '-').repeat(n - 1) + '*');


  //under middle
  for (let i = 1; i <= n - 2; i++) {
    console.log(' '.repeat(i) + ('*' + '-').repeat(n - i - 1) + '*' + ' '.repeat(i));
  }
  //bottom
  console.log(' '.repeat(n - 1) + '*' + ' '.repeat(n - 1));
}

//perfectDiamond(['5']);
//perfectDiamond(['1']);

File: Pyrhon - Introduction to Programming/8.1. Exam Preparation - Part I/09. Perfect Diamond.py
n = int(input())

if n == 1:
    print(' ' * (n - 1) + '*' + ' ' * (n - 1))
    exit()

#top
print(' ' * (n - 1) + '*' + ' ' * (n - 1))

#before middle
[print(' ' * (n - i - 1) + ('*' + '-') * i + '*' + ' ' * (n - i - 1)) for i in range(1, n - 1)]

#middle
print(('*' + '-') * (n - 1) + '*')

#under middle
[print(' ' * i + ('*' + '-') * (n - i - 1) + '*' + ' ' * i) for i in range(1, n - 1)]

#bottom
print(' ' * (n - 1) + '*' + ' ' * (n - 1))


'''
3
  *  
 *-* 
*-*-*
 *-* 
  *  
'''
